Signal/src/call/UserInterface/CallUIAdapter.swift

```diff
@@ -145,15 +145,17 @@ extension CallUIAdaptee {
         adaptee.setIsMuted(call: call, isMuted: isMuted)
     }
 
-    internal func setHasVideo(call: SignalCall, hasVideo: Bool) {
-        adaptee.setHasVideo(call: call, hasVideo: hasVideo)
+    internal func setHasLocalVideo(call: SignalCall, hasLocalVideo: Bool) {
+        adaptee.setHasLocalVideo(call: call, hasLocalVideo: hasLocalVideo)
     }
 
-    internal func toggleSpeakerphone(call: SignalCall, isEnabled: Bool) {
+    internal func setIsSpeakerphoneEnabled(call: SignalCall, isEnabled: Bool) {
         // Speakerphone is not handled by CallKit (e.g. there is no CXAction), so we handle it w/o going through the 
         // adaptee, relying on the AudioService CallObserver to put the system in a state consistent with the call's 
         // assigned property.
-        call.isSpeakerphoneEnabled = isEnabled
+        CallService.signalingQueue.async {
+            call.isSpeakerphoneEnabled = isEnabled
+        }
```

Comment:
This is a good example of one of the reasons I'm uncomfortable with how signaling queue is being used - it's easy to forget that CallService and SignalCall state should only be modified by the signaling queue.